test/libraries/Order.sol

```diff
@@ -17,6 +17,59 @@ library Order {
         bool partiallyFillable;
     }
 
+    // I wish I could declare the following as constants and export them as part
+    // of the library. However, "Only constants of value type and byte array
+    // type are implemented." and "Library cannot have non-constant state
+    // variables". So I'm left with defining them as functions.
+
+    function ALL_KINDS() internal pure returns (bytes32[2] memory) {
+        return [GPv2Order.KIND_SELL, GPv2Order.KIND_BUY];
+    }
+
+    function ALL_SELL_TOKEN_BALANCES() internal pure returns (bytes32[3] memory) {
+        return [GPv2Order.BALANCE_ERC20, GPv2Order.BALANCE_EXTERNAL, GPv2Order.BALANCE_INTERNAL];
+    }
+
+    function ALL_BUY_TOKEN_BALANCES() internal pure returns (bytes32[2] memory) {
+        return [GPv2Order.BALANCE_ERC20, GPv2Order.BALANCE_INTERNAL];
+    }
+
+    function ALL_FLAGS() internal pure returns (Flags[] memory out) {
+        uint256 numBools = 2;
```

Comment:
I'm a little confused as to "numBools" of 2, yet Flags only contains a single boolean parameter?

Reply:
On further inspection, it seems this is referring to the number of permutations associated with boolean values (and given a single boolean, would correspond to 2 permutations).

Given this, I'd sooner see `numBools = 1` and then in the array length declared below, to multiply `numBools` by 2, with a comment on the `out` variable reflecting that it contains all permutations of all flags.

Reply:
Good point. Is it better now?